service/geopmdpy/system_files.py

```diff
@@ -813,7 +822,11 @@ def start_profile(self, client_pid, profile_name):
         self.check_client_active(client_pid, 'start_profile')
         if 'profile_name' in self._sessions[client_pid]:
             raise RuntimeError(f'Client pid {client_pid} has requested profiling twice')
-        uid, gid, _ = self._pid_info(client_pid)
+        try:
+            uid, gid, _ = self._pid_info(client_pid)
+        except psutil.NoSuchProcess:
+            sys.stderr.write('Warning: Profile cannot be created for PID {client_pid}, it is no longer active')
```

Comment:
The only way I can see to hit this is if the client_pid dies between the check_client_active() call on 813 and here.
If the client_pid is dead here, whose responsibility is it to call remove_client() to get it out of self._sessions?